cloud/pkg/admissioncontroller/mutate_offlinemigration.go

```diff
@@ -49,28 +44,25 @@ func mutateOfflineMigration(review admissionv1beta1.AdmissionReview) *admissionv
 }
 
 func generatePatch(tolerations []corev1.Toleration) []patchMapValue {
-	patch := []patchMapValue{{
-		Op:   "add",
-		Path: "/spec/template/spec/tolerations",
-		Value: []map[string]string{{
-			"key":      corev1.TaintNodeUnreachable,
-			"operator": Exists,
-			"effect":   NoExecute,
-		}},
-	}}
-	if len(tolerations) > 0 {
-		for _, toleration := range tolerations {
-			if toleration.Key == corev1.TaintNodeUnreachable {
-				if toleration.Effect == "NoExecute" &&
-					toleration.Operator == "Exists" && toleration.TolerationSeconds == nil {
-					return nil
-				}
-				toleration.TolerationSeconds = nil
-				patch[0].Op = "replace"
-			}
+	currentTolerations := make([]interface{}, 0, len(tolerations)+1)
+	for _, v := range tolerations {
+		if v.Key == corev1.TaintNodeUnreachable {
+			continue
 		}
+		currentTolerations = append(currentTolerations, v)
+	}
```

Comment:
If toleration key is `TaintNodeUnreachable` and effect is `NoSchdule`, do we need to keep it here?

Reply:
```
currentTolerations = append(currentTolerations, corev1.Toleration{
	Key:      corev1.TaintNodeUnreachable,
	Operator: corev1.TolerationOpExists,
	Effect:   corev1.TaintEffectNoExecute,
})
```
@fisherxu  If toleration key is TaintNodeUnreachable, We use custom toleration content.

Reply:
@fisherxu I think the `effect` can be directly covered into empty to make it adapt to all the effects.

Reply:
Could we update the `Path: "/spec/template/spec/tolerations",` in another pr first? And leave the key and effect for more discussuion. :)

Reply:
There is no need to open another PR. Except for `path`, other logical results are the same as before

Reply:
The difference is whether care about `NoSchdule`? Now in the patch only add the `NoExecute`.

Reply:
You can refer to the old version, which is also the default `NoExecution`.
More discussuion, I can open another PR.